src/index.js

```diff
@@ -1477,13 +1457,42 @@ export class ULabel {
         this.update_annotation_mode();
         this.update_current_class();
 
+        // Update class counter
+        this.toolbox.redraw_update_items(this);
+
         // Set transparancy for inactive layers
         this.readjust_subtask_opacities();
 
         // Redraw demo
         this.redraw_demo();
     }
 
+    /**
+     * Switch to the next subtask in the toolbox
+     */
+    switch_to_next_subtask() {
+        let current_subtask = this.state["current_subtask"];
+        let toolbox_tab_keys = [];
+
+        // Put all of the toolbox tab keys in a list
+        for (let idx in this.toolbox.tabs) {
+            toolbox_tab_keys.push(this.toolbox.tabs[idx].subtask_key);
+        }
+
+        // Get the index of the next subtask in line
+        let new_subtask_index = toolbox_tab_keys.indexOf(current_subtask) + 1;  // +1 gets the next subtask
+
+        // If the current subtask was the last one in the array, then
+        // loop around to the first subtask
+        if (new_subtask_index === toolbox_tab_keys.length) {
+            new_subtask_index = 0;
+        }
+
+        let new_subtask = toolbox_tab_keys[new_subtask_index];
+
+        this.set_subtask(new_subtask);
```

Comment:
```suggestion
        let current_subtask = this.state["current_subtask"];

        let new_subtask_index = this.toolbox.tabs.findIndex(
            tab => tab.subtask_key === current_subtask
        ) + 1;

        if (new_subtask_index === this.toolbox.tabs.length) {
            new_subtask_index = 0;
        }

        this.set_subtask(
            this.toolbox.tabs[new_subtask_index].subtask_key
        );
```
`findIndex` to simplify logic and readability.

I know you were just moving existing code into a function but I couldn't help it

Reply:
[0221a2d](https://github.com/SenteraLLC/ulabel/pull/163/commits/0221a2dfcdcd799ed53ca00491cb9e05a992acdb)